### PR TITLE
Make outputs of CLI examples nice

### LINKED ISSUE #905: Make outputs of CLI examples nice.
The CLI output is a bit messy and this makes it difficult to read.

I suggest putting everything into tables and to reduce the amount of data in the `list` commands and to add additional commands to see more details about a particular item.

We can use a dependency such as https://docs.rs/tabled/latest/tabled/ to achieve this.

### PR DESCRIPTION
**Pull request by LagginTimes**
Closes #905

Initial work done towards #905 using https://docs.rs/tabled/latest/tabled/.

The following commands are now in table form:
```
address index
address list
balance
txout list
```

### CODE REVIEW

**Review by evanlinjin — APPROVED** [review]
I think this is objectively better than what we had before. Any further improvements can be done later imo.

**Review by remix7531 — COMMENTED** [review]
Looks a lot better. However, this make it a lot harder to grep output.

bdk-table

**Comment by notmandatory** [comment]
I agree with @remix75 this output format looks good but makes it hard to parse. In many of our existing tutorials with the old `bdk-cli` we use `jq` to parse the json output. If we made these tables the default output style can we also add (in a separate PR is OK) a new `--json` flag (or something like that) to format command outputs as json ?
